packages/preview/not-tudabeamer-2023/0.1.0/lib.typ

```diff
@@ -0,0 +1,178 @@
+    block(
+      height: 1.99in - self.page-args.margin.top,
+      below: 0.24in,
+      width: 100% - 2in,
+      align(bottom)[#heading(level: 2, upper(title))]
+    )
+    body
+  }, ..args)
+}
+
+#let title-slide(self: none, ..args) = {
+  self.enable-header = false
+  let info = self.info + args.named()
+  (self.methods.touying-slide)(self: self, repeat: none, setting: body => {
+    grid(
+      columns: 100%,
+      rows: (
+        2.76in - self.page-args.margin.top,
+        1.18in,
+        1.6in
+      ),
+      gutter: (0in, 0.2in),
+      grid.cell([]),
+      grid.cell(align: bottom + center, title-font(upper(info.title))),
+      grid.cell(align: top + center, subtitle-font(info.subtitle)),
+    )
+    body
+  }, ..args)
+}
+
+#let new-section-slide(self: none, section, ..args) = {
+  (self.methods.touying-slide)(self: self, repeat: none, section: section, setting: body => {
+    grid(
+      columns: 100%,
+      rows: (
+        4.32in - self.page-args.margin.top,
+        1.18in,
+        1.18in
+      ),
+      gutter: (0in, 0.05in),
+      grid.cell([]),
+      grid.cell(align: bottom, heading(level: 2, upper(states.current-section-with-numbering(self)))),
+      grid.cell(align: top, subtitle-font([]))
+    )
+    body
+  }, ..args)
+}
+
+#let outline-slide(self: none, leading: 50pt) = {
+  (self.methods.slide)(self: self, repeat: none, title: "Contents", (self.methods.touying-outline)(self: self))
+}
+
+#let d-outline(self: none, enum-args: (:), list-args: (:), cover: true) = states.touying-progress-with-sections(dict => {
+  let (current-sections, final-sections) = dict
+  current-sections = current-sections.filter(section => section.loc != none)
+  final-sections = final-sections.filter(section => section.loc != none)
+  let current-index = current-sections.len() - 1
+  set enum(..enum-args)
+  set enum(numbering: n => title-font[#n], spacing: 0.4in, body-indent: 0.25in)
+  show enum: set align(horizon)
+  v(2.91in-1.18in-self.page-args.margin.top)
+  columns(2,
+    for (i, section) in final-sections.enumerate() {
+    {
+        enum.item(i + 1, [#link(section.loc, section.title)<touying-link>])
+      }
+      parbreak()
+  })
+})
+
+#let slides(self: none, title-slide: true, outline-slide: true, slide-level: 1, ..args) = {
+  if title-slide {
+    (self.methods.title-slide)(self: self)
+  }
+  if outline-slide {
+    (self.methods.outline-slide)(self: self)
+  }
+  (self.methods.touying-slides)(self: self, slide-level: slide-level, ..args)
+}
+
+#let register(
+  self: themes.default.register(),
+  tuda_logo,
+  ..args,
+) = {
+  self = (self.methods.datetime-format)(self: self, "[day].[month].[year]")
+  self.enable-header = true
+  let header = self => align(top)[
+    #if self.enable-header {
+      v(0.39in)
+      header-font(upper(self.info.short-title + " / " + self.info.short-author))
+    }
+    #place(top + right, dx: 0.34in, dy: 0.2in)[#image.decode(tuda_logo, height: 0.99in)]
```

Comment:
Maybe it would be better here for it to show any logo content instead of an image. This way we can also use emoji as a logo. You can refer to the practice of the touying stargazer theme. Of course, this is not necessary, but it will be more user-friendly.